File: cassandrablog/src/main/webapp/app/entities/cassandrablog/cass-add-ons-selected-by-organization/cass-add-ons-selected-by-organization.test-samples.ts
import dayjs from 'dayjs/esm';

import { ICassAddOnsSelectedByOrganization, NewCassAddOnsSelectedByOrganization } from './cass-add-ons-selected-by-organization.model';

export const sampleWithRequiredData: ICassAddOnsSelectedByOrganization = {
  compositeId: {
    organizationId: 'sample-organizationId-1',
    arrivalDate: dayjs('2024-01-02T12:00:00Z'),
    accountNumber: 'sample-accountNumber-1',
    createdTimeId: 'sample-createdTimeId-1',
  },
};

export const sampleWithPartialData: ICassAddOnsSelectedByOrganization = {
  compositeId: {
    organizationId: 'sample-organizationId-2',
    arrivalDate: dayjs('2024-01-03T12:00:00Z'),
    accountNumber: 'sample-accountNumber-2',
    createdTimeId: 'sample-createdTimeId-2',
  },
  departureDate: dayjs('2024-01-03T12:00:00Z'),
  customerId: 'sample-customerId-2',
  customerFirstName: 'sample-customerFirstName-2',
  customerLastName: 'sample-customerLastName-2',
  customerUpdatedEmail: 'sample-customerUpdatedEmail-2',
  customerUpdatedPhoneNumber: 'sample-customerUpdatedPhoneNumber-2',
};

export const sampleWithFullData: ICassAddOnsSelectedByOrganization = {
  compositeId: {
    organizationId: 'sample-organizationId-3',
    arrivalDate: dayjs('2024-01-04T12:00:00Z'),
    accountNumber: 'sample-accountNumber-3',
    createdTimeId: 'sample-createdTimeId-3',
  },
  departureDate: dayjs('2024-01-04T12:00:00Z'),
  customerId: 'sample-customerId-3',
  customerFirstName: 'sample-customerFirstName-3',
  customerLastName: 'sample-customerLastName-3',
  customerUpdatedEmail: 'sample-customerUpdatedEmail-3',
  customerUpdatedPhoneNumber: 'sample-customerUpdatedPhoneNumber-3',
  customerEstimatedArrivalTime: 'sample-customerEstimatedArrivalTime-3',
  tinyUrlShortCode: 'sample-tinyUrlShortCode-3',
  addOnDetailsText: { key3: 'val-3' },
  addOnDetailsDecimal: { key3: 3 },
  addOnDetailsBoolean: { key3: false },
  addOnDetailsBigInt: { key3: dayjs('2024-01-04T12:00:00Z') },
};

export const sampleWithNewData: NewCassAddOnsSelectedByOrganization = {
  compositeId: {
    organizationId: 'sample-organizationId-4',
    arrivalDate: dayjs('2024-01-05T12:00:00Z'),
    accountNumber: 'sample-accountNumber-4',
    createdTimeId: 'sample-createdTimeId-4',
  },
};

Object.freeze(sampleWithNewData);
Object.freeze(sampleWithRequiredData);
Object.freeze(sampleWithPartialData);
Object.freeze(sampleWithFullData);

File: cassandrablog/src/test/java/com/saathratri/developer/cassandra/blog/domain/CassAddOnsSelectedByOrganizationTestSamples.java
package com.saathratri.developer.cassandra.blog.domain;

import java.util.Random;
import java.util.UUID;
import java.util.concurrent.atomic.AtomicLong;

public class CassAddOnsSelectedByOrganizationTestSamples {

    private static final Random random = new Random();
    private static final AtomicLong longCount = new AtomicLong(random.nextInt() + (2 * Integer.MAX_VALUE));

    public static CassAddOnsSelectedByOrganization getCassAddOnsSelectedByOrganizationSample1() {
        return new CassAddOnsSelectedByOrganization()
            .compositeId(
                new CassAddOnsSelectedByOrganizationId()
                    .organizationId(UUID.fromString("23d8dc04-a48b-45d9-a01d-4b728f0ad4aa"))
                    .arrivalDate(1L)
                    .accountNumber("accountNumber1")
                    .createdTimeId(UUID.fromString("23d8dc04-a48b-45d9-a01d-4b728f0ad4aa"))
            )
            .departureDate(1L)
            .customerId(UUID.fromString("23d8dc04-a48b-45d9-a01d-4b728f0ad4aa"))
            .customerFirstName("customerFirstName1")
            .customerLastName("customerLastName1")
            .customerUpdatedEmail("customerUpdatedEmail1")
            .customerUpdatedPhoneNumber("customerUpdatedPhoneNumber1")
            .customerEstimatedArrivalTime("customerEstimatedArrivalTime1")
            .tinyUrlShortCode("tinyUrlShortCode1")
            .addOnDetailsText(
                new java.util.HashMap<String, String>() {
                    {
                        put("addOnDetailsText1", "addOnDetailsText1");
                    }
                }
            )
            .addOnDetailsBigInt(
                new java.util.HashMap<String, Long>() {
                    {
                        put("addOnDetailsBigInt1", 1L);
                    }
                }
            );
    }

    public static CassAddOnsSelectedByOrganization getCassAddOnsSelectedByOrganizationSample2() {
        return new CassAddOnsSelectedByOrganization()
            .compositeId(
                new CassAddOnsSelectedByOrganizationId()
                    .organizationId(UUID.fromString("ad79f240-3727-46c3-b89f-2cf6ebd74367"))
                    .arrivalDate(2L)
                    .accountNumber("accountNumber2")
                    .createdTimeId(UUID.fromString("ad79f240-3727-46c3-b89f-2cf6ebd74367"))
            )
            .departureDate(1L)
            .customerId(UUID.fromString("23d8dc04-a48b-45d9-a01d-4b728f0ad4aa"))
            .customerFirstName("customerFirstName1")
            .customerLastName("customerLastName1")
            .customerUpdatedEmail("customerUpdatedEmail1")
            .customerUpdatedPhoneNumber("customerUpdatedPhoneNumber1")
            .customerEstimatedArrivalTime("customerEstimatedArrivalTime1")
            .tinyUrlShortCode("tinyUrlShortCode1")
            .addOnDetailsText(
                new java.util.HashMap<String, String>() {
                    {
                        put("addOnDetailsText1", "addOnDetailsText1");
                    }
                }
            )
            .addOnDetailsBigInt(
                new java.util.HashMap<String, Long>() {
                    {
                        put("addOnDetailsBigInt1", 1L);
                    }
                }
            );
    }

    public static CassAddOnsSelectedByOrganization getCassAddOnsSelectedByOrganizationRandomSampleGenerator() {
        return new CassAddOnsSelectedByOrganization()
            .compositeId(
                new CassAddOnsSelectedByOrganizationId()
                    .organizationId(UUID.randomUUID())
                    .arrivalDate(longCount.incrementAndGet())
                    .accountNumber(UUID.randomUUID().toString())
                    .createdTimeId(UUID.randomUUID())
            )
            .departureDate(1L)
            .customerId(UUID.fromString("23d8dc04-a48b-45d9-a01d-4b728f0ad4aa"))
            .customerFirstName("customerFirstName1")
            .customerLastName("customerLastName1")
            .customerUpdatedEmail("customerUpdatedEmail1")
            .customerUpdatedPhoneNumber("customerUpdatedPhoneNumber1")
            .customerEstimatedArrivalTime("customerEstimatedArrivalTime1")
            .tinyUrlShortCode("tinyUrlShortCode1")
            .addOnDetailsText(
                new java.util.HashMap<String, String>() {
                    {
                        put("addOnDetailsText1", "addOnDetailsText1");
                    }
                }
            )
            .addOnDetailsBigInt(
                new java.util.HashMap<String, Long>() {
                    {
                        put("addOnDetailsBigInt1", 1L);
                    }
                }
            );
    }
}
